modules/oai.xql

```diff
@@ -0,0 +1,135 @@
+declare namespace response="http://exist-db.org/xquery/response";
+declare namespace util="http://exist-db.org/xquery/util";
+declare namespace oai="http://www.openarchives.org/OAI/2.0/";
+
+import module namespace functx="http://www.functx.com";
+import module namespace crud="http://xquery.weber-gesamtausgabe.de/modules/crud" at "crud.xqm";
+import module namespace config="http://xquery.weber-gesamtausgabe.de/modules/config" at "config.xqm";
+import module namespace wega-util="http://xquery.weber-gesamtausgabe.de/modules/wega-util" at "wega-util.xqm";
+import module namespace date="http://xquery.weber-gesamtausgabe.de/modules/date" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/date.xqm";
+import module namespace str="http://xquery.weber-gesamtausgabe.de/modules/str" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/str.xqm";
+import module namespace mycache="http://xquery.weber-gesamtausgabe.de/modules/cache" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/cache.xqm";
+import module namespace lod="http://xquery.weber-gesamtausgabe.de/modules/lod" at "lod.xqm";
+import module namespace lang="http://xquery.weber-gesamtausgabe.de/modules/lang" at "lang.xqm";
+import module namespace query="http://xquery.weber-gesamtausgabe.de/modules/query" at "query.xqm";
+import module namespace wdt="http://xquery.weber-gesamtausgabe.de/modules/wdt" at "wdt.xqm";
+
+declare option output:method "xml";
+declare option output:media-type "application/xml";
+declare option output:indent "yes";
+
+(:~
+ : Get the last date of modification from dataHistory.xml. Fallback: VersionDate from options.xml
+ : @author Dennis Ried 
+ :
+ : @param $docID The ID of the document
+ : return The date ad xs:dateTime or empty 
+:)
+declare %private function oai:last-modified($docID) as xs:dateTime {
+    let $props := config:get-data-props($docID)
+    return
+        if($props?dateTime castable as xs:dateTime) 
+        then ($props?dateTime => xs:dateTime())
+        else (fn:current-dateTime())
+};
+
+(:~
+ : Create a header response
+ :
+ : @author Dennis Ried 
+:)
+declare %private function oai:response-headers($docID) as empty-sequence() {
+    response:set-header('Access-Control-Allow-Origin', '*'),
+    response:set-header('Last-Modified', date:rfc822(oai:last-modified($docID))), 
+    response:set-header('Cache-Control', 'max-age=300,public')
+};
+
+(:~
+ : Creating the response for the interface (header, calling record by oai:record)
+ :
+ : @author Dennis Ried 
+:)
+declare function oai:oai($model as map(*)) as node() {
+	<OAI-PMH xmlns="http://www.openarchives.org/OAI/2.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/ http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd">
+		 <responseDate>{fn:current-dateTime()}</responseDate>
+		 <request verb="GetRecord" identifier="{$lod-metadata?DC.identifier}" metadataPrefix="oai_dc">http://www.openarchives.org/OAI/2.0/oai_dc/</request> 
```

Comment:
The variable `$lod-metadata` is not available in this function. It's probably best to fetch `lod:metadata` first and append this to the main `$model` (see below)?! 

```suggestion
		 <request verb="GetRecord" identifier="{$model?lod?DC.identifier}" metadataPrefix="oai_dc">http://www.openarchives.org/OAI/2.0/oai_dc/</request>
```

Reply:
I understand, but why does it work on my machine?